playbook/app/pb_kits/playbook/pb_radio/_radio.tsx

```diff
@@ -42,52 +41,103 @@ const Radio = ({
   value = 'radio_text',
   onChange = () => { void 0 },
   ...props
-}: RadioProps, ref: any) => {
-  const ariaProps = buildAriaProps(aria)
-  const dataProps = buildDataProps(data)
-  const htmlProps = buildHtmlProps(htmlOptions)
+}: RadioProps ) => {
+  const radioRef = useRef(null);
+
+  const ariaProps = buildAriaProps(aria);
+  const dataProps = buildDataProps(data);
+  const htmlProps = buildHtmlProps(htmlOptions);
   const classes = classnames(
-    buildCss('pb_radio_kit', alignment ),
-    dark ? 'dark': null, error ? 'error': null,
+    buildCss('pb_radio_kit', alignment),
+    dark ? 'dark' : null,
+    error ? 'error' : null,
     globalProps(props),
-    className)
+    className
+  );
+
+  const classesCustom = classnames(
+    dark ? 'dark' : null,
+    error ? 'error' : null,
+    globalProps(props),
+    className
+  );
+
+  const isCustomChild = children && isValidElement(children) && children.type !== 'input';
 
   const displayRadio = (props: RadioProps & any) => {
-    if (children)
-      return (children)
-    else
-    return (
-    <input
-        disabled={disabled}
-        id={id}
-        name={name}
-        onChange={onChange}
-        ref={ref}
-        text={text}
-        type="radio"
-        value={value}
-        {...props}
-    />
-  )}
+    if (isValidElement(children) && children.type === 'input') {
+      return React.cloneElement(children, { ...props, ref: radioRef });
+    } else if (isCustomChild || !children) {
+      return (
+        <input
+            disabled={disabled}
+            id={id}
+            name={name}
+            onChange={onChange}
+            ref={radioRef}
+            text={text}
+            type="radio"
+            value={value}
+            {...props}
+        />
+      );
+    }
+  };
+
+  const handleContainerClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent> | undefined) => {
+    if (event) {
+      const target = event.target as HTMLElement;
+      if (
+        target.id === 'children-wrapper' ||
```

Comment:
Looks good! i would maybe change this id name to be a bit more specific, just because 'children-wrapper' feels too generic for Nitro. Maybe something like 'pb-radio-children-wrapper' ?

Reply:
great call! changing that now thank you!